File: apps/otel-rust/src/otel.rs
// Copyright (c) Nicolas Lamirault <nicolas.lamirault@gmail.com>
//
// SPDX-License-Identifier: Apache-2.0

mod resource_provider;
pub use resource_provider::create_resource;

mod otel_logger;
pub use otel_logger::init_logger;

mod otel_meter;
pub use otel_meter::init_meter;

mod otel_tracer;
pub use otel_tracer::init_tracer;

File: apps/otel-python/otelpython/telemetry/otel.py
# SPDX-FileCopyrightText: Copyright (C) Nicolas Lamirault <nicolas.lamirault@gmail.com>
# SPDX-License-Identifier: Apache-2.0

from opentelemetry import metrics
from opentelemetry import trace
from opentelemetry.sdk import resources

from otelpython import settings
from otelpython import version


def create_resource(service_name: str) -> resources.Resource:
    return resources.Resource.create(
        {
            resources.SERVICE_NAME: service_name,
            resources.SERVICE_VERSION: version.version_info,
        }
    )


def get_tracer() -> trace.Tracer:
    """Returns the OpenTelemetry tracer instance."""
    return trace.get_tracer(settings.OTEL_SERVICE_NAME)


def get_meter() -> metrics.Meter:
    """Returns the OpenTelemetry meter instance."""
    return metrics.get_meter(settings.OTEL_SERVICE_NAME)
